src/compiler/checker.ts

```diff
@@ -38424,6 +38392,45 @@ namespace ts {
             }
         }
 
+        function checkUnmatchedJSDocParameters(node: SignatureDeclaration) {
+            const jsdocParameters = filter(getJSDocTags(node), isJSDocParameterTag);
+            if (length(jsdocParameters) === 0) return;
+
+            const isJs = isInJSFile(node);
+            const parameters = new Set<__String>();
+            const excludedParameters = new Set<number>();
+            forEach(node.parameters, ({ name }, index) => {
+                if (isIdentifier(name)) {
+                    parameters.add(name.escapedText);
+                }
+                if (isBindingPattern(name)) {
+                    excludedParameters.add(index);
+                }
+            });
+
+            const containsArguments = containsArgumentsReference(node);
+            if (containsArguments) {
+                const lastJSDocParam = lastOrUndefined(jsdocParameters);
+                if (lastJSDocParam && isIdentifier(lastJSDocParam.name) && lastJSDocParam.typeExpression &&

```

Comment:
the old code works with Identifiers and QualifiedNames; this should too (and we should have a test for it if there's not one)

Reply:
@sandersn There are tests [`paramTagNestedWithoutTopLevelObject`](https://github.com/microsoft/TypeScript/blob/main/tests/cases/conformance/jsdoc/paramTagNestedWithoutTopLevelObject.ts), [`paramTagNestedWithoutTopLevelObject2`](https://github.com/microsoft/TypeScript/blob/main/tests/cases/conformance/jsdoc/paramTagNestedWithoutTopLevelObject2.ts), [`paramTagNestedWithoutTopLevelObject3`](https://github.com/microsoft/TypeScript/blob/main/tests/cases/conformance/jsdoc/paramTagNestedWithoutTopLevelObject3.ts), [`paramTagNestedWithoutTopLevelObject4`](https://github.com/microsoft/TypeScript/blob/main/tests/cases/conformance/jsdoc/paramTagNestedWithoutTopLevelObject4.ts). The _baseline_ of these tests hasn't changed, I suppose the error handling for _QualifiedNames_ works as expected.